### DISCUSSION TITLE
include a file based on environment variables such as "include ${HOSTNAME}-font.conf"

### DISCUSSION THREAD

**Discussion by benibilme** [kovidgoyal/kitty#6991] [Q&A] [answered]
I am trying to use same kitty.conf for different computers. I have created platform specific files which the common kitty.conf files include. I use a script to generate necessary symbolic links to platform specific files 

For example in kitty.conf 

````
include font.conf
````

A script  links the ${HOSTNAME}-font.conf to font.conf  such as font.conf  etc.  

I really do not want to use a script to do that. Is there a way to that in kitty.conf. I know that `env` directive. But it seems it does not honor $HOSTNAME env variable. 

What I would like to do is following: 

````
include ${HOSTNAME}-font.conf
````

However above command fails. 

Any help much appreciated.

**Comment by kovidgoyal** [comment]
include ${HOSTNAME}.conf

works fine but you need to make sure HOSTNAME is defined in whatever
environment you are running kitty in. In particular setting it in your
shell rc files wont work, since those run *after* kitty starts.

**Comment by benibilme** [comment]
Thank you,

I have not defined `$HOSTNAME` in any of my `rc` files. It is available by default.  In my terminal, I can `echo $HOSTNAME` in kitty terminal bash shell,  but somehow kitty itself does not see it. I checked via `map f1 show_kitty_env_vars`. 

I added a different variable for not messing with the system settings such as  `PLATFORM=$(hostnamectl hostname)`  in  `.bash_profile`, and bound  `$PLATFORM` variable in `kitty.conf`  as  `include ${PLATFORM}-font.conf`, now it works. I do not know why `$HOSTNAME` does not propagate to kitty.  

I wish such a command `include $(hostnamectl hostname)-font.conf `   would be possible in `kitty.conf` , which would be far cleaner solution for me.

**Reply by kovidgoyal** [reply, marked as answer]
Just because it isnt in your personal rc files doesnt mean its not in one of the system ones. Allowing command execution in conf files makes parsing and loading config slow and non-static. It isnt really worth the tradeoff. If you really want something like that simply make a wrapper script to run kitty in that script use whatever commands you like to generate your config and either directly pass it to kitty via --config or use --override

**Comment by Luflosi** [comment]
Also see https://github.com/kovidgoyal/kitty/issues/811.
